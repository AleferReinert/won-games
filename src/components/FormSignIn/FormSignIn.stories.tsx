import type { StoryObj, Meta } from '@storybook/react'
import { within } from '@storybook/testing-library'
import { expect } from '@storybook/jest'
import FormSignInComponent from './FormSignIn'

const meta: Meta<typeof FormSignInComponent> = {
  title: 'Components/FormSignIn',
  component: FormSignInComponent,
  parameters: {
    backgrounds: {
      default: 'Light'
    }
  },
  decorators: [
    (Story) => (
      <div style={{ maxWidth: '38rem', margin: '0 auto' }}>
        <Story />
      </div>
    )
  ]
}
export default meta

type Story = StoryObj<typeof FormSignInComponent>

export const FormSignIn: Story = {
  play: ({ canvasElement }) => {
    const canvas = within(canvasElement)
    const email = canvas.getByPlaceholderText(/e-mail/i)
    const password = canvas.getByPlaceholderText(/password/i)
    const forgotYourPassword = canvas.getByRole('link', {
      name: /forgot your password?/i
    })
    const signInNow = canvas.getByRole('button', { name: /sign in now/i })
    const dontHaveAccount = canvas.getByText(/don't have an account?/i)
    const signUp = canvas.getByRole('link', { name: /sign up/i })

    expect(email).toBeInTheDocument()
    expect(password).toBeInTheDocument()
    expect(forgotYourPassword).toBeInTheDocument()
    expect(signInNow).toBeInTheDocument()
    expect(dontHaveAccount).toBeInTheDocument()
    expect(signUp).toBeInTheDocument()
  }
}
