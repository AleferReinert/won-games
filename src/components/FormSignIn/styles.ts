import styled, { css } from 'styled-components'
import * as TextFieldStyles from 'components/TextField/styles'
import { darken } from 'polished'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    ${TextFieldStyles.Wrapper} {
      margin-bottom: ${theme.spacings.xxsmall};
    }
  `}
`

export const ForgotPassword = styled.a`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xsmall};
    color: ${theme.colors.black};
    text-align: right;
    margin-bottom: ${theme.spacings.medium};
    display: block;
    text-decoration: none;
  `}
`

export const SignLink = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.black};
    text-align: center;
    margin-top: ${theme.spacings.xsmall};

    a {
      color: ${theme.colors.secondary};
      margin-left: ${theme.spacings.xxsmall};
      transition: color, border, ${theme.transition.fast};
      border-bottom: 0.1rem solid ${theme.colors.secondary};
      text-decoration: none;

      &:hover {
        color: ${darken(0.1, theme.colors.secondary)};
        border-bottom-color: ${darken(0.1, theme.colors.secondary)};
      }
    }
  `}
`
