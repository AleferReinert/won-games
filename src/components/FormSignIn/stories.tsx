import { ComponentStory, ComponentMeta } from '@storybook/react'
import FormSignIn from '.'

export default {
  title: 'Forms/FormSignIn',
  component: FormSignIn,
  parameters: {
    backgrounds: {
      default: 'Light'
    }
  }
} as ComponentMeta<typeof FormSignIn>

export const Default: ComponentStory<typeof FormSignIn> = (args) => (
  <div style={{ width: '38rem', margin: '0 auto' }}>
    <FormSignIn {...args} />
  </div>
)
