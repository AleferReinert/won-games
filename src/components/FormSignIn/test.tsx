import { screen } from '@testing-library/react'
import FormSignIn from '.'
import { renderWithTheme } from 'utils/tests/helpers'

describe('<FormSignIn />', () => {
  it('should render a sign in form', () => {
    const { container } = renderWithTheme(<FormSignIn />)

    expect(screen.getByPlaceholderText(/e-mail/i)).toBeInTheDocument()
    expect(screen.getByPlaceholderText(/password/i)).toBeInTheDocument()
    expect(
      screen.getByRole('link', {
        name: /forgot your password\?/i
      })
    ).toBeInTheDocument()
    expect(screen.getByText(/don't have an account\?/i)).toBeInTheDocument()
    expect(
      screen.getByRole('link', {
        name: /sign up/i
      })
    ).toBeInTheDocument()
    expect(
      screen.getByRole('button', { name: /sign in now/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
