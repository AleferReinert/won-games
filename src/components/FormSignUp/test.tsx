import { screen } from '@testing-library/react'
import FormSignIn from '.'
import { renderWithTheme } from 'utils/tests/helpers'

describe('<FormSignUp />', () => {
  it('should render a sign up form', () => {
    const { container } = renderWithTheme(<FormSignIn />)

    expect(screen.getByPlaceholderText(/name/i)).toBeInTheDocument()
    expect(screen.getByPlaceholderText(/e-mail/i)).toBeInTheDocument()
    expect(screen.getByPlaceholderText('Password')).toBeInTheDocument()
    expect(screen.getByPlaceholderText('Confirm password')).toBeInTheDocument()
    expect(screen.getByText(/already have an account\?/i)).toBeInTheDocument()
    expect(
      screen.getByRole('link', {
        name: /sign in/i
      })
    ).toBeInTheDocument()
    expect(
      screen.getByRole('button', { name: /sign up now/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
