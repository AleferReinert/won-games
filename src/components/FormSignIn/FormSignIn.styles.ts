import styled, { css } from 'styled-components'
import { FormWrapperStyle, FormLinkStyle } from 'templates/Auth/Auth.styles'

export const FormWrapper = styled.div`
  ${FormWrapperStyle}
`

export const FormLink = styled.div`
  ${FormLinkStyle}
`

export const ForgotPassword = styled.a`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xsmall};
    color: ${theme.colors.black};
    text-align: right;
    margin-bottom: ${theme.spacings.medium};
    display: block;
    text-decoration: none;
  `}
`
