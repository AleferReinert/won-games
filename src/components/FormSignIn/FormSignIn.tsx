import { Email, Lock } from '@styled-icons/material-outlined'
import TextField from 'components/TextField/TextField'
import Link from 'next/link'
import Button from 'components/Button/Button'
import * as S from './FormSignIn.styles'

const FormSignIn = () => {
  return (
    <S.FormWrapper>
      <form>
        <TextField
          name='email'
          type='email'
          placeholder='E-mail'
          icon={<Email />}
        />
        <TextField
          name='password'
          type='password'
          placeholder='Password'
          icon={<Lock />}
        />
        <S.ForgotPassword href='/link'>Forgot your password?</S.ForgotPassword>
        <Button full size='large'>
          Sign in now
        </Button>
        <S.FormLink>
          Don&apos;t have an account?
          <Link href='/sign-up'>Sign up</Link>
        </S.FormLink>
      </form>
    </S.FormWrapper>
  )
}

export default FormSignIn
