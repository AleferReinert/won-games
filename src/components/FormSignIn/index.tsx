import TextField from 'components/TextField'
import * as S from './styles'
import { Email, Lock } from '@styled-icons/material-outlined'
import Button from 'components/Button'
import Link from 'next/link'

const FormSignIn = () => {
  return (
    <S.Wrapper>
      <form>
        <TextField
          name='email'
          type='email'
          placeholder='E-mail'
          icon={<Email />}
        />
        <TextField
          name='password'
          type='password'
          placeholder='Password'
          icon={<Lock />}
        />
        <S.ForgotPassword href='/link'>Forgot your password?</S.ForgotPassword>
        <Button fullWidth size='large'>
          Sign in now
        </Button>
        <S.SignLink>
          Don&apos;t have an account?
          <Link href='/sign-up'>Sign up</Link>
        </S.SignLink>
      </form>
    </S.Wrapper>
  )
}

export default FormSignIn
