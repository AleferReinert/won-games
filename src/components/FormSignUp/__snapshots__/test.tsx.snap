// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormSignUp /> should render a sign up form 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: #EAEAEA;
  border-radius: 0.2rem;
  padding: 0 0.8rem;
  border: 0.2rem solid;
  border-color: #EAEAEA;
}

.c2:focus-within {
  box-shadow: 0 0 0.5rem #F231A5;
}

.c5 {
  color: #030517;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 1.6rem;
  padding: 0.8rem;
  background: transparent;
  border: 0;
  outline: none;
  width: 100%;
}

.c5:-webkit-autofill {
  -webkit-box-shadow: 0 0 0px 1000px #EAEAEA inset;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 2rem;
  color: #8F8F8F;
  margin: 0 0.8rem;
}

.c3 > svg {
  width: 100%;
}

.c0 .c1 {
  margin-bottom: 0.8rem;
}

.c7 {
  font-size: 1.4rem;
  color: #030517;
  text-align: center;
  margin-top: 1.6rem;
}

.c7 a {
  color: #3CD3C1;
  margin-left: 0.8rem;
  -webkit-transition: color,border,0.1s ease-in-out;
  transition: color,border,0.1s ease-in-out;
  border-bottom: 0.1rem solid #3CD3C1;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c7 a:hover {
  color: #29b3a3;
  border-bottom-color: #29b3a3;
}

.c4 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c6 {
  background: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  color: #FAFAFA;
  cursor: pointer;
  border: 0;
  border-radius: 0.4rem;
  padding: 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 5rem;
  font-size: 1.6rem;
  padding: 0 4.0rem;
  width: 100%;
}

.c6:hover {
  background: linear-gradient(180deg,#e35565 0%,#d958a6 50%);
}

<div
  class="c0"
>
  <form>
    <div
      class="c1 "
    >
      <div
        class="c2"
      >
        <span
          class="c3"
        >
          <svg
            aria-hidden="true"
            class="c4"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM7.35 18.5C8.66 17.56 10.26 17 12 17s3.34.56 4.65 1.5c-1.31.94-2.91 1.5-4.65 1.5s-3.34-.56-4.65-1.5zm10.79-1.38a9.947 9.947 0 0 0-12.28 0A7.957 7.957 0 0 1 4 12c0-4.42 3.58-8 8-8s8 3.58 8 8c0 1.95-.7 3.73-1.86 5.12z"
            />
            <path
              d="M12 6c-1.93 0-3.5 1.57-3.5 3.5S10.07 13 12 13s3.5-1.57 3.5-3.5S13.93 6 12 6zm0 5c-.83 0-1.5-.67-1.5-1.5S11.17 8 12 8s1.5.67 1.5 1.5S12.83 11 12 11z"
            />
          </svg>
        </span>
        <input
          autocomplete="off"
          class="c5"
          name="name"
          placeholder="Name"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="c1 "
    >
      <div
        class="c2"
      >
        <span
          class="c3"
        >
          <svg
            aria-hidden="true"
            class="c4"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M22 6c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6zm-2 0-8 5-8-5h16zm0 12H4V8l8 5 8-5v10z"
            />
          </svg>
        </span>
        <input
          autocomplete="off"
          class="c5"
          name="email"
          placeholder="E-mail"
          type="email"
          value=""
        />
      </div>
    </div>
    <div
      class="c1 "
    >
      <div
        class="c2"
      >
        <span
          class="c3"
        >
          <svg
            aria-hidden="true"
            class="c4"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
            />
          </svg>
        </span>
        <input
          autocomplete="off"
          class="c5"
          name="password"
          placeholder="Password"
          type="password"
          value=""
        />
      </div>
    </div>
    <div
      class="c1 "
    >
      <div
        class="c2"
      >
        <span
          class="c3"
        >
          <svg
            aria-hidden="true"
            class="c4"
            fill="currentColor"
            focusable="false"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
            />
          </svg>
        </span>
        <input
          autocomplete="off"
          class="c5"
          name="confirm-password"
          placeholder="Confirm password"
          type="password"
          value=""
        />
      </div>
    </div>
    <button
      class="c6"
    >
      <span>
        Sign up now
      </span>
    </button>
    <div
      class="c7"
    >
      Already have an account?
      <a
        href="/sign-in"
      >
        Sign in
      </a>
    </div>
  </form>
</div>
`;
